impeller/renderer/backend/gles/render_pass_gles.cc

```diff
@@ -309,6 +309,10 @@ void RenderPassGLES::ResetGLState(const ProcTableGLES& gl) {
     }
   }
 
+  CullMode current_cull_mode = CullMode::kNone;
+  WindingOrder current_winding_order = WindingOrder::kClockwise;
+  gl.FrontFace(GL_CW);
```

Comment:
Can you pull these into helper functions like this:
```c++
void SetFrontFace(ProcTable& gl, 
    std::optional<WindingOrder>& cached_winding_order, 
    WindingOrder new_winding_order) {
  if (cached_winding_order.has_value() && *cached_winding_order == new_winding_order) {
    return;
  }
  cached_winding_order = new_winding_order;
  gl.FrontFace(foobar(new_winding_order));
}
```

Reply:
I can pull them into closures, but this whole thing is a big static function so there isn't anywhere else to store the fields. WDUT?

Reply:
The closures don't really clean things up much.  I was thinking the helper functions would be helpful abstractions over setting only something if it is different or isn't set.  The cached values are stored on the stack.  It's not a huge deal.